src/hooks/useUser.ts

```diff
@@ -0,0 +1,44 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { clearStoredUser, getStoredUser, setStoredUser } from './userStorage';
+import { snsApiClient } from '~/api';
+
+async function getUser(user: any) {
+  if (!user) return null;
+
+  const { data } = await snsApiClient.get('/auth-user', {
+    headers: {
+      Authorization: `Bearer ${user.token}`
+    }
+  });
+  return { user: data, token: user.token };
+}
+
+function useUser() {
+  const queryClient = useQueryClient();
+
+  const { data: user } = useQuery({
+    queryKey: ['user'],
+    queryFn: (): any => getUser(user),
+    initialData: getStoredUser(),
+    onSuccess: (received) => {
+      if (received) {
+        setStoredUser(received);
+      } else {
+        clearStoredUser();
+      }
```

Comment:
삼항 연산자를 사용해도 괜찮을 것 같아요!

```suggestion
      received ? setStoredUser(received) : clearStoredUser()
```

Reply:
아... 코드 쉽지 않아서 정신없이 따라하다보니 화살표 함수 사용을 잊었네요.... 🙇‍♀️🙇‍♀️🙇‍♀️